### ISSUE TITLE
agent preserve the same nodeID after reinstall

### ISSUE THREAD

**Issue by smartekIT** [Ylianst/MeshCentral#3678]
HI,
It happened recently when MS pushed for Windows 11 updates and Anti-virus software getting updated with new rules for reputation that some of the agents get uninstalled or just deleted it's root folder in these computers affected by this update rollup.

the issue is when we try to re-install the agent back again on those machines, it register itself on the server with a new NodeID which will actually create a big issue for us as many other settings and services been configured to use the old NodeID to identify that machine specially RouterCMD services.

the question is, can we some how overwrite the NodeID for an agent so it doesn't get a new one but instead it takes the NodeID we assigned which obviously our old agent NodeID. this way all the settings and notes for that machine stays the same ?

also in the server side, i know there is a way to move an agent to different group, but can we actually force an agent to change it's NodeID to specified one which is our old NodeID we have all the settings and notes on ?

Thanks heaps

**Comment by smartekIT** [comment]
if this can't be done, how about adding some tag recognize in the RouterCMD, currently we map in RouterCMD using MeshID and NodeID. but if we use MeshID and Special Tag detection would be an alternative.
for e.g. we can apply one special tag in that agent settings on the server side, and inside RouterCMD mapping settings file we could use that tag in addition to the MeshID for the group while the special tag searching for that device inside that group.

**Comment by krayon007** [comment]
So in the past, I've seen sometimes during windows updates, the certificate in the windows cert store was lost, which caused the agent to generate a new certificate, which causes the node Id to change.

If the certificate is intact, and only the agent binary is lost (db is preserved), if you reinstall the agent without performing an uninstall, the certificate in the cert store will be used, which makes it so the agent node id remains the same.

There isn't a way for an agent to be "assigned" a nodeid, because the node id is something the agent needs to cryptographically prove to the server.

**Comment by smartekIT** [comment]
thank you @krayon007 for clarifying this, but when the agent was removed by windows, I noticed all files including binary and db were removed from the installation directory. which I guess leads to loose that NodeID for good.
based on  what you clarified, I guess best would be if we some how try the plan B which I suggested earlier with RouterCMD mapping to have an option to use agent special tag. This will save us lots for time and efforts trying to reach out to users to update their RouterCMD settings whenever the NodeID get lost.

RouterCMD was a brilliant idea and to be honest it's one of the best option I've seen in the whole project. since @Ylianst introduced it with multiple mapping option then it becomes everyday use for us and saved us a lot of time.

so RouterCMD with ability to map to agent with MeshID and Tag instead or in addition to the current method which is MeshID and NodeID.
